### ISSUE TITLE
sort_by() to support multiple prioritized keys

### ISSUE THREAD

**Issue by eripa** [mikefarah/yq#1541]
**Please describe your feature request.**

It would be great to have way to impose second, third etc sorting order.  See the below example. I.e. first sort all documents on X, but for if the value is the same, use the second key Y to sort within those.

Related to #1024

https://github.com/mikefarah/yq/issues/1024#issuecomment-1413025115

**Describe the solution you'd like**
If we have `file.yml` like:

```yaml
name: Beta
other: Epsilon
---
name: Alfa
other: Gamma
---
name: Beta
other: Delta
```

And we run a command:

```bash
yq ea '[.] | sort_by(.name, .other) | .[] | splitDoc' file.yaml
```

it could output

```yaml
name: Alfa
other: Gamma
---
name: Beta
other: Delta
---
name: Beta
other: Epsilon
```

**Describe alternatives you've considered**
Have not tried any other solutions. As far as I know this isn't supported, see: https://github.com/mikefarah/yq/issues/1024#issuecomment-1413025115

**Comment by mikefarah** [comment]
Fixed in v4.31.1, see https://mikefarah.gitbook.io/yq/operators/sort#sort-by-multiple-fields

**Comment by eripa** [comment]
This is awesome, thank you!
